platform_design/lib/main.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'tab_1.dart';
+import 'tab_2.dart';
+import 'tab_3.dart';
+import 'tab_4.dart';
+import 'widgets.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  final platformOverride = ValueNotifier<TargetPlatform>(defaultTargetPlatform);
+
+  @override
+  Widget build(BuildContext context) {
+    // Change this value to better see animations.
+    timeDilation = 1;
+    // Either Material or Cupertino widgets work in either Material or Cupertino
+    // Apps.
+    return ValueListenableBuilder(
+      valueListenable: platformOverride,
+      child: PlatformAdaptingHomePage(platformOverride: platformOverride),
+      builder: (BuildContext context, TargetPlatform platform, Widget child) {
+        return MaterialApp(
+          title: 'Adaptive Music App',
+          theme: ThemeData(
+            // Use the green theme for Material widgets.
+            primarySwatch: Colors.green,
+            platform: platform,
+          ),
+          builder: (BuildContext context, Widget child) {
+            return CupertinoTheme(
+              // Instead of letting Cupertino widgets auto-adapt to the Material
+              // theme (which is green), we're going to use a different theme
+              // for Cupertino (which is blue by default).
+              data: CupertinoThemeData(),
+              child: Material(child: child),
+            );
+          },
+          home: child,
+        );
+      },
+    );
+  }
+}
+
+// The widget which shows a different type of scaffold depending on the platform.
+//
+// This file has the most amount of non-sharable code since it behaves the most
+// differently between the platforms.
+class PlatformAdaptingHomePage extends StatefulWidget {
+  PlatformAdaptingHomePage({ Key key, this.platformOverride }) : super(key: key);
+
+  final ValueNotifier<TargetPlatform> platformOverride;
+
+  @override
+  _PlatformAdaptingHomePageState createState() => _PlatformAdaptingHomePageState();
+}
+
+class _PlatformAdaptingHomePageState extends State<PlatformAdaptingHomePage> {
+  // We're keeping global keys because tabs 1 and 2 own a bunch of data.
```

Comment:
It's less so for efficiency and more so to make it visually apparent that just the chrome is changing. 

What happens otherwise is that we re-generate everything in tab 1 and all the colors of the cards change again. It might make people think we're pulling wool over their eyes and that everything's changing whereas we're just changing the chrome. 

It's mostly for a glanceable visual effect. We can remove the tab2 key though

Reply:
went for a single key for just tab 1